src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -129,33 +267,437 @@ public ResultSet getSchemas() throws SQLException {
 
     @Override
     public ResultSet getCatalogs() throws SQLException {
-        throw new SQLFeatureNotSupportedException("TODO");
+        MongoJsonSchema resultSchema = MongoJsonSchema.createEmptyObjectSchema();
+        resultSchema.addRequiredScalarKeys(new Pair<>(TABLE_CAT, BSON_STRING_TYPE_NAME));
+
+        // All fields in this result set are nested under the bottom namespace.
+        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
+        botSchema.properties.put(BOT_NAME, resultSchema);
+
+        BsonExplicitCursor c =
+                new BsonExplicitCursor(
+                        this.conn
+                                .mongoClient
+                                .listDatabaseNames()
+                                .into(new ArrayList<>())
+                                .stream()
+                                .sorted()
+                                .map(
+                                        dbName ->
+                                                createBottomBson(
+                                                        new BsonElement(
+                                                                TABLE_CAT, new BsonString(dbName))))
+                                .collect(Collectors.toList()));
+
+        return new MongoSQLResultSet(null, c, botSchema);
+    }
+
+    /**
+     * liftSQLException tries to execute a Supplier that may throw a RuntimeException that wraps a
+     * SQLException as the cause. If such an exception is encountered, the inner SQLException is
+     * thrown. Otherwise, the Supplier executes as expected.
+     *
+     * <p>This method is intended for use with Stream API methods that wrap SQLExceptions in
+     * RuntimeExceptions to appease the compiler. We want to propagate those SQLExceptions all the
+     * way out to the caller so this method is a way of recovering them.
+     *
+     * @param f The Supplier function to execute. This function may throw a RuntimeException that
+     *     wraps a SQLException.
+     * @param <T> The return type of the Supplier.
+     * @return The return value of the Supplier.
+     * @throws SQLException If a RuntimeException that wraps a SQLException is caught.
+     */
+    private static <T> T liftSQLException(Supplier<T> f) throws SQLException {
+        try {
+            return f.get();
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof SQLException) {
+                throw (SQLException) e.getCause();
+            }
+            throw e;
+        }
+    }
+
+    // Helper class for representing all info needed to serialize column data for the
+    // getColumns and getColumnPrivileges methods. Intended for use with toGetColumnsDoc
+    // and toGetColumnPrivilegesDoc helpers.
+    private static class GetColumnsDocInfo {
+        String dbName;
+        String collName;
```

Comment:
I understand why this is named "collName" since it is the collection name, but I must tell you that I got confused for a while at why you were applying a table name pattern to collName. I was reading colName and thought it was the column name. I guess once you are aware of this, it's easy enough to remember and check if it is collName or columnName, but would renaming this to collectionName or tableName look like a suitable option for you? For me, it will save us the trouble to have to be extra vigilant on that one and when you'll get new member in the team it will also make ramp-up easier because there will be no potential for error.

Reply:
Done